src/common/utils/registry.h

```diff
@@ -16,6 +16,83 @@
 
 namespace registry
 {
+    namespace install_scope
+    {
+        const wchar_t INSTALL_SCOPE_REG_KEY[] = L"Software\\Classes\\powertoys\\";
+
+        enum class InstallScope
+        {
+            PerMachine = 0,
+            PerUser,
+        };
+
+        inline const InstallScope get_current_install_scope()
+        {
+            // Open HKLM key
+            HKEY perMachineKey{};
+            if (RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                              INSTALL_SCOPE_REG_KEY,
+                              0,
+                              KEY_READ,
+                              &perMachineKey) != ERROR_SUCCESS)
+            {
+                // Open HKCU key
+                HKEY perUserKey{};
+                if (RegOpenKeyExW(HKEY_CURRENT_USER,
+                                  INSTALL_SCOPE_REG_KEY,
+                                  0,
+                                  KEY_READ,
+                                  &perUserKey) != ERROR_SUCCESS)
+                {
+                    // both keys are missing
+                    return InstallScope::PerMachine;
+                }
+                else
+                {
+                    DWORD dataSize{};
+                    if (RegGetValueW(
+                        perUserKey,
+                        nullptr,
+                        L"InstallScope",
+                        RRF_RT_REG_SZ,
+                        nullptr,
+                        nullptr,
+                        &dataSize) != ERROR_SUCCESS)
+                    {
+                        // HKCU key is missing
+                        RegCloseKey(perUserKey);
+                        return InstallScope::PerMachine;
+                    }
+
+                    std::wstring data;
+                    data.resize(dataSize / sizeof(wchar_t));
+
+                    if (RegGetValueW(
+                            perUserKey,
+                            nullptr,
+                            L"InstallScope",
+                            RRF_RT_REG_SZ,
+                            nullptr,
+                            &data[0],
+                            &dataSize) != ERROR_SUCCESS)
+                    {
+                        // HKCU key is missing
+                        RegCloseKey(perUserKey);
+                        return InstallScope::PerMachine;
```

Comment:
What makes more sense?
'No HKCU'=Machine or 'HKLM exist'=Machine. What happens if both are installed because a user tries to go around company restrictions for update cycles?

Reply:
> 'HKLM exist'=Machine

This is not possible as we need to be backward compatible with the versions that doesn't have this reg entry.

> What makes more sense?
'No HKCU'=Machine or 'HKLM exist'=Machine. What happens if both are installed because a user tries to go around company restrictions for update cycles?

This is yet to be decided, i.e. what to do in this scenario

Reply:
>> What makes more sense?
'No HKCU'=Machine or 'HKLM exist'=Machine. What happens if both are installed because a user tries to go around company restrictions for update cycles?

> This is yet to be decided, i.e. what to do in this scenario

Fyi:
1. Admins expect that machine GPOs have priority over user GPOs. It the normal behavior in Windows, Office, Edge, ...
2. If a user has no admin permission he can't manipulate user and machine policies. Neither using gpedit.msc, nor changing the Registry values.